keps/sig-scheduling/5278-nominated-node-name-for-expectation/README.md

```diff
@@ -0,0 +1,781 @@
+
+So, here we just give a hint of a reasonable SLO, you need to adjust it based on your cluster's usual behaviors.
+
+In the default scheduler, we should see the throughput around 100-150 pods/s ([ref](https://perf-dash.k8s.io/#/?jobname=gce-5000Nodes&metriccategoryname=Scheduler&metricname=LoadSchedulingThroughput&TestName=load)), and this feature shouldn't bring any regression there.
+
+Based on that: 
+- `schedule_attempts_total` shouldn't be less than 100 in a second.
+- the average of `scheduling_algorithm_duration_seconds` shouldn't be above 10 ms.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [x] Metrics
+  - `schedule_attempts_total` with `scheduled` label.
+  - `scheduler_pod_scheduling_sli_duration_seconds` with `scheduled` label.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+No.
+
+### Dependencies
+
+No.
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+Yes.
+- API call type: PATCH pods.
+- estimated throughput: Each pod that goes through Permit or PreBind plugins triggers one additional API call.
+In the default scheduler, pods with DRA or delayed binding PVC would be those.
+- originating component: Kube-scheduler
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+Yes - but it should be negligible impact.
+The memory usage in kube-scheduler is supposed to increase by external components starting to use this
+because when `NominatedNodeName` is added on the pods, the scheduler's internal component called `nominator` has to record them so that scheduling cycles can refer to them as necessary.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+The scheduler itself doesn't work anymore in that case.
+
+###### What are other known failure modes?
+
+Unknown.
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+## Implementation History
+
+- 7th May 2025: The initial KEP is submitted.
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+## Alternatives
```

Comment:
Two questions, that I'm not sure where to post:
- How `NominatedNodeName` plays with `NodeName`, can we ensure this is explicitly pointed out in this document? 
- Have you considered adding a PodSpec `NominatedNodeName` to cover the use-case Wojtek mentioned to clearly separate nomination (status) from hint (spec)? I'd imagine this being at least explored in the alternatives section with explicit reason why it was rejected.

Reply:
1) NNN is treated as "current plan of record", but something that is not necessarily a final decision.   NodeName is the actual binding decision.

2) Yes - it was explored and we decided that distinguishing that for now is not really needed and can be done in an additive way in the future. So let's not add the API until we are really sure it's needed.